hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/SparkAdapterSupport.scala

```diff
@@ -26,17 +26,6 @@ import org.apache.spark.sql.hudi.SparkAdapter
  */
 trait SparkAdapterSupport {
 
-  lazy val sparkAdapter: SparkAdapter = {
```

Comment:
Instead of moving this to Java let's dot he following:

- Create companion object `ScalaAdapterSupport` 
- Move this conditional there
- Keep this var (for compatibility) referencing static one from the object